### PR TITLE
Retry to get templates from ACR

### PR DESCRIPTION
**Pull request by QuanWanxx**

### CODE REVIEW

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/ContainerRegistry/ContainerRegistryTemplateProvider.cs [hunk 1]
```diff
@@ -60,7 +61,15 @@ public class ContainerRegistryTemplateProvider : IContainerRegistryTemplateProvi
             try
             {
                 var provider = _templateCollectionProviderFactory.CreateTemplateCollectionProvider(schemaImageReference, accessToken);
-                return await provider.GetTemplateCollectionAsync(cancellationToken);
+
+                // "TemplateManagementException" is base exception for "ContainerRegistryAuthenticationException" and "ImageFetchException"
+                return await Policy
+                  .Handle<TemplateManagementException>()
+                  .RetryAsync(3, onRetry: (exception, retryCount) =>
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/ContainerRegistry/ContainerRegistryTemplateProvider.cs:68]: Shall we use exponential back-off for the retry logic?

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataFilter/ContainerRegistryFilterProvider.cs [hunk 2]
```diff
@@ -62,7 +63,13 @@ public async Task<FilterConfiguration> GetFilterAsync(CancellationToken cancella
             {
                 AcrClient client = new AcrClient(imageInfo.Registry, accessToken);
                 var filterConfigurationProvider = new OciArtifactProvider(imageInfo, client);
-                var acrImage = await filterConfigurationProvider.GetOciArtifactAsync(cancellationToken);
+                var acrImage = await Policy
+                  .Handle<TemplateManagementException>()
+                  .RetryAsync(3, onRetry: (exception, retryCount) =>
+                  {
+                      _logger.LogWarning(exception, "Failed to get image from Container Registry. Retry {RetryCount}.", retryCount);
```
Comment (moria97) [inline FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataFilter/ContainerRegistryFilterProvider.cs:70]: Maybe we can add the image reference to the logs? like `Failed to get image {reference} from container registry.

Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/DataFilter/ContainerRegistryFilterProvider.cs]: Added, thanks.

File: FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/ContainerRegistry/ContainerRegistryTemplateProvider.cs (hunk 1 above)
Comment (QuanWanxx) [reply FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/ContainerRegistry/ContainerRegistryTemplateProvider.cs]: Added, thanks.

**Review by moria97 — APPROVED** [review]

**Pull request merged**